File: 백준/Gold/2565. 전깃줄/전깃줄.cc
#include <iostream>
#include <vector>
#include <algorithm>

#define FASTIO() ios_base::sync_with_stdio(false), cin.tie(NULL), cout.tie(NULL)

using namespace std;

int N;

struct p
{
    int left;
    int right;

    p(int left, int right) : left(left), right(right) {}
    p() {}

    bool const operator<(const p temp) const
    {
        return left < temp.left;
    }
};

vector<p> v;
int dp[505];

int main()
{
    FASTIO();
    cin >> N;

    for (int i = 0; i < N; i++)
    {
        int x, y;
        cin >> x >> y;

        v.push_back(p(x, y));
    }

    sort(v.begin(), v.end());

    for (int i = 0; i < 505; i++)
    {
        dp[i] = 1;
    }

    int answer = 0;

    for (int i = 0; i < N; i++)
    {
        for (int j = 0; j < i; j++)
        {
            if (v[i].right > v[j].right)
            {
                dp[i] = max(dp[i], dp[j] + 1);
            }
        }
        if (dp[i] > answer)
        {
            answer = dp[i];
        }
    }

    cout << N - answer << "\n";
}

File: 백준/Gold/2565. 전깃줄/전깃줄.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {

	public static class Point implements Comparable<Point> {

		public final int left;
		public final int right;

		public Point(int left, int right) {
			this.left = left;
			this.right = right;
		}

		@Override
		public int compareTo(Point point) {
			return this.left - point.left;
		}
	}

	static Integer[] dp;
	static Point[] arr;

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int N = in.nextInt();

		dp = new Integer[505];
		arr = new Point[N];

		for (int i = 0; i < N; i++) {
			int x = in.nextInt();
			int y = in.nextInt();

			arr[i] = new Point(x, y);
		}

		for (int i = 0; i < N; i++) {
			dp[i] = 1;
		}

		Arrays.sort(arr);

		int answer = 0;

		for (int i = 1; i < N; i++) {
			for (int j = 0; j < i; j++) {
				if (arr[i].right > arr[j].right) {
					dp[i] = Math.max(dp[i], dp[j] + 1);
				}
			}

			if (dp[i] > answer)
				answer = dp[i];
		}

		System.out.println(N - answer);

	}

}
